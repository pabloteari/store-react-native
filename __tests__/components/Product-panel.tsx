import 'react-native';
import React from 'react';
import ProductPanel from '../../src/components/product-panel';

import renderer from 'react-test-renderer';
jest.mock('react-native-vector-icons/EvilIcons', () => 'EvilIcons');

it('product panel render correctly', () => {
  renderer.create(
    <ProductPanel
      id={'1'}
      name={'name'}
      image={'image'}
      stock={0}
      price={'price'}
      createdAt={'createAt'}
      count={0}
    />,
  );
});

test('renders tree product correctly', () => {
  const tree = renderer
    .create(
      <ProductPanel
        id={'1'}
        name={'name'}
        image={'image'}
        stock={0}
        price={'price'}
        createdAt={'createAt'}
        count={0}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
