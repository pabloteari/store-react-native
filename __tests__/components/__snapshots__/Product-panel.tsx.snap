// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tree correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginVertical": 5,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 10,
          }
        }
      >
        Invalid date
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        price
      </Text>
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 10,
          }
        }
      >
        left
         
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders tree product correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginVertical": 5,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 20,
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 10,
          }
        }
      >
        Invalid date
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        price
      </Text>
      <Text
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 10,
          }
        }
      >
        left
         
        0
      </Text>
    </View>
  </View>
</View>
`;
